tools/build_addons.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+from concurrent.futures import ThreadPoolExecutor
+
+ROOT_DIR = os.path.abspath(os.path.join(__file__, '..', '..'))
+
+def install_and_rebuild(args, install_args):
+  out_dir = os.path.abspath(args.out_dir)
+  node_bin = os.path.join(out_dir, 'node')
+  if args.is_win:
+    node_bin += '.exe'
+
+  if os.path.isabs(args.node_gyp):
+    node_gyp = args.node_gyp
+  else:
+    node_gyp = os.path.join(ROOT_DIR, args.node_gyp)
+
+  # Create a temporary directory for node headers.
+  with tempfile.TemporaryDirectory() as headers_dir:
```

Comment:
Does this reinstall the headers every time the addons are changed (but the headers aren't)? Note that the addons are also rebuilt when we update addons.md and we don't distinguish between pure documentation changes and code example changes. It would be better to install them at a fixed location under `out/${build_type}/` and only reinstall when the header changes. Then `test/addons/.buildstamp` can depend on it and we pass the installation path to `build_addons.py`.

Reply:
I added a change in `Makefile` to make the node headers only re-generate when there is header file changed.